### ISSUE TITLE
FMDB Queue Leaking Memory when doing Consecutive transactions in DB

### ISSUE THREAD

**Issue by nshetty26** [ccgus/fmdb#484]
I was running into an issue where FMDB was leaking massive amounts of memory (150mb on an iPad Mini every time I ran my sync routine) 

code in question:
```swift
self.queue.inDatabase({[dbMessages = messages](db: FMDatabase!) -> Void in
            db.setKey(DbWrapper.dbKey)
            db.setShouldCacheStatements(false)
            db.beginTransaction()
            for dbMessage in dbMessages {
            let success = db.executeUpdate("INSERT or IGNORE INTO Entities (EntityID, Type) Values (?, ?)", withArgumentsInArray: [dbMessage.entityID, dbMessage.type])
                
                if success {
                    db.executeUpdate("UPDATE Entities SET PersonID = ?, JSON = ?, TimeStamp = ?, Dirty = ?, Type = ? where EntityID = ? AND Type = ?", withArgumentsInArray: [dbMessage.patientID, dbMessage.jsonData, NSDate(), dbMessage.dirty, dbMessage.type, dbMessage.entityID, dbMessage.type])
                }
            }
            db.commit()
})
```

I did my due diligence and checked everywhere around FMDBQueue and once i switched from the Queue to the following code:
```swift
        if db.open() {
            db.setKey(DbWrapper.dbKey)
            db.setShouldCacheStatements(false)
            db.beginTransaction()
            for message in messages {
                let success = db.executeUpdate("INSERT or IGNORE INTO Entities (EntityID, Type) Values (?, ?)", withArgumentsInArray: [message.entityID, message.type])
                
                if success {
                    db.executeUpdate("UPDATE Entities SET PersonID = ?, JSON = ?, TimeStamp = ?, Dirty = ?, Type = ? where EntityID = ? AND Type = ?", withArgumentsInArray: [message.patientID, message.jsonData, NSDate(), message.dirty, message.type, message.entityID, message.type])
                }
            }
            db.commit()
            db.close()
}
```
All my memory leak issues were fixed.

Some other information:
1. Everything is processed Serially using NSOperations and an OperationQueue
2. My sync operation gets and inserts all the rows at a per patient level and can be anywhere between 10 - 400 inserts done in a transaction
3. our DB Schema is 5 columns but one of those a JSON blob turned into NSData before we put it into the DB.

More than happy to answer any other questions but after 3 days of working on this the leak was happening inside FMDBQueue when mass transactions are happened at a rapid fire rate.  We never experienced anything like this when we were only inserting 1-40 rows at once every 20-60 seconds.

**Comment by ccgus** [comment]
I'm curious- if you don't call db.close(), does the leak still happen?

**Comment by nshetty26** [comment]
The leak does not happen when I dont use db.close, it definitely causes some unintended behavior around sleeps and the DB being locked but memory is not leaked.

**Comment by ccgus** [comment]
What do you mean around sleeps? Are you using the db ivar from multiple threads?

**Comment by nshetty26** [comment]
I was able to trace the bug to iOS version 9.1.  The leak happens during the scenario above but it is caused by the OS not deallocating properly.  This leak is fixed in version 9.3.  Again, no leak in Queue directly and am able to keep my code in tact to use FMDB queue.
